core/common/src/Utf8.kt

```diff
@@ -66,26 +66,25 @@
  * </tr>
  * </table>
  */
-@file:JvmName("Utf8")
 
 package kotlinx.io
 
-import kotlin.jvm.JvmName
-import kotlin.jvm.JvmOverloads
+import kotlinx.io.internal.*
 
 /**
- * Returns the number of bytes used to encode the slice of `string` as UTF-8 when using
- * [Sink.writeUtf8].
+ * Returns the number of bytes used to encode the slice of `string` as UTF-8 when using [Sink.writeUtf8].
+ *
+ * @param startIndex the index (inclusive) of the first character to encode, `0` by default.
+ * @param endIndex the index (exclusive) of the character past the last character to encode, `string.length` by default.
+ *
+ * @throws IndexOutOfBoundsException when [startIndex] or [endIndex] is out of range of string indices.
+ * @throws IllegalArgumentException when `startIndex > endIndex`.
  */
-@JvmOverloads
-@JvmName("size")
-fun String.utf8Size(beginIndex: Int = 0, endIndex: Int = length): Long {
-  require(beginIndex >= 0) { "beginIndex < 0: $beginIndex" }
-  require(endIndex >= beginIndex) { "endIndex < beginIndex: $endIndex < $beginIndex" }
-  require(endIndex <= length) { "endIndex > string.length: $endIndex > $length" }
+public fun String.utf8Size(startIndex: Int = 0, endIndex: Int = length): Long {
```

Comment:
Done

Reply:
I was actually relying on this function in Krossbow, because I need to truncate a string based on its length in UTF-8 bytes (not characters). The use case is the "close reason" (the text payload) of the web socket `Close` frame. As per the [specification](https://datatracker.ietf.org/doc/html/rfc6455#section-5.5), the payload of all control frames (Close, Ping, Pong) must be <= 125 bytes.

Without this function, I need to encode the string into a byte array, then check if it's too long, and then truncate. With the `utf8Size` function, I could optimize the happy path by not copying bytes around if the string was short enough.

Reply:
I opened an issue for this https://github.com/Kotlin/kotlinx-io/issues/258. I figured it's a better place than here. Sorry for the noise